maintainer/feature_maturity_level.yaml

```diff
@@ -0,0 +1,91 @@
+# Definition of feature: a feature is a plugin or a registry.
+#
+# Feature maturity grading rules:
+# 1. A feature's maturity level is `experimental` when its implementation is complete.
+# 2. A feature is `stable` when it meets both of the following conditions:
+#   2.1 It has been in a release version for more than 6 months.
+#   2.2 There are no bugfixes on this feature in the last three months.
+# 3. If a breaking change occurs or the maintainer thinks it is not stable enough, it returns to `experimental`.
+#
+# When creating new release, please update the maturity level of the features in this file, according to
+# the above rules.
+#
+# When adding a new feature, please set the experimental_since field to the next release.
+
+plugins:
+  - name: bandwidth_limit
+    status: experimental
+    experimental_since: 0.4.0
```

Comment:
I suggest us to add a file for maturity level for every plugin and compile them when build the plugin, so we can:

1. Get log information (if we need) to show the maturity level.
2. Leverage git history to record the version of maturity.
3. Update the plugin itself for maturity improvement with the file inside, instead of just updating some readme outside.
4. Add more information in the file in the future per need (e.g. vendor, contributor, website, category, etc)

Reply:
For every plugin, the plugin's doc already shows its own maturity level. We can use it to record the change of maturity across different versions.
There is no plan to maintain a plugin's metadata file as there is no mechanism to consume such metadata file. As our resources are limited, designing such a metadata file looks premature.